pkg/scheduler/framework/runtime/registry_test.go

```diff
@@ -56,3 +58,108 @@ func TestRegistry_Filter(t *testing.T) {
 		})
 	}
 }
+
+func TestRegistry_Register(t *testing.T) {
+	var r = make(Registry)
+
+	tests := []struct {
+		name              string
+		pluginFactoryName string
+		wantErr           bool
+		expectedPlugins   []string
+	}{
+		{
+			name:              "p1-1",
+			pluginFactoryName: "p1",
+			wantErr:           false,
+			expectedPlugins:   []string{"p1"},
+		},
+		{
+			name:              "p2-1",
+			pluginFactoryName: "p2",
+			wantErr:           false,
+			expectedPlugins:   []string{"p1", "p2"},
+		},
+		{
+			name:              "p1-2",
+			pluginFactoryName: "p1",
+			wantErr:           true,
+			expectedPlugins:   []string{"p1", "p2"},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := r.Register(tt.pluginFactoryName, mockPluginFactory)
+
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Register() error = %v, wantErr %v", err, tt.wantErr)
+			}
+
+			if !reflect.DeepEqual(r.FactoryNames(), tt.expectedPlugins) {
```

Comment:
We can't just `deepEqual` two slices here, because the `first slice` is dumped from a map, and the item is unordered.

Reply:
Eh, I copied this logic from original test. I will fix this.

Reply:
I found out the `FactoryNames()` already did the sort operation, so there is no need to sort by ourselves.

Reply:
Fine. But a little bit tricky because we mush ensure the test data is orderd.